src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeTracker.cs

```diff
@@ -77,8 +94,10 @@ public void StartFileChangeListeningAsync()
                 return newCookie;
             }, LazyThreadSafetyMode.ExecutionAndPublication);
 
-            // file change service is free-threaded. start running it in background right away
-            Task.Run(() => _fileChangeCookie.Value, CancellationToken.None);
+            lock (s_lastBackgroundTaskGate)
+            {
+                s_lastBackgroundTask = s_lastBackgroundTask.ContinueWith(_ => _fileChangeCookie.Value, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default);
```

Comment:
instead of a task per _fileCHangeCookie.Value, this feels like one of those areas where we should kick the can, and batch up a set of calls.  However, this change seems like a reasonable thing to do, esp if we're just going to replace this all with a new API.